backend/service/k8s/node.go

```diff
@@ -0,0 +1,70 @@
+package k8s
+
+import (
+	"context"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	k8sapiv1 "github.com/lyft/clutch/backend/api/k8s/v1"
+)
+
+func (s *svc) DescribeNode(ctx context.Context, clientset, cluster, name string) (*k8sapiv1.Node, error) {
+	// Node is a cluster level object, so namespace should be ""
+	cs, err := s.manager.GetK8sClientset(ctx, clientset, cluster, metav1.NamespaceAll)
+	if err != nil {
+		return nil, err
+	}
+
+	nodes, err := cs.CoreV1().Nodes().List(ctx, metav1.ListOptions{
+		FieldSelector: "metadata.name=" + name,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(nodes.Items) == 1 {
+		return ProtoForNode(cs.Cluster(), &nodes.Items[0]), nil
+	} else if len(nodes.Items) > 1 {
+		return nil, status.Error(codes.FailedPrecondition, "located multiple nodes")
```

Comment:
nit: might be nice to include the name in this response message

Reply:
Similar to the previous comment, this matches the message for other types, ex. pods https://github.com/lyft/clutch/blob/main/backend/service/k8s/pods.go#L32. I kept it the same for uniformity. Tbh this condition should basically never be true as kubernetes names are unique per object types (but may be reused - either way only 1 will ever exist at a time) https://github.com/lyft/clutch/blob/main/backend/service/k8s/pods.go#L32